[PATCH] Tests fails to compile with libnpupnp

```
[119/123] /usr/bin/c++ -DATRAILERS -DBSD -DBSD_NATIVE_UUID -DFMT_SHARED -DHAVE_CURL -DHAVE_INOTIFY -DHAVE_JS -DHAVE_NL_LANGINFO -DHAVE_SETLOCALE -DONLINE_SERVICES -DPACKAGE_DATADIR=\"/usr/local/share/gerbera\" -DPACKAGE_NAME=\"Gerbera\" -DSOPCAST -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -DUSING_NPUPNP -DVERSION=\"1.6.1\" -I/usr/local/include/npupnp -I/usr/ports/net/gerbera/work/gerbera-1.6.1/src -O2 -pipe -march=ivybridge -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -Wall -O2 -pipe -march=ivybridge -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -pthread -std=c++17 -MD -MT test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o -MF test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o.d -o test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o -c /usr/ports/net/gerbera/work/gerbera-1.6.1/test/util/test_upnp_headers.cc
FAILED: test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o
/usr/bin/c++ -DATRAILERS -DBSD -DBSD_NATIVE_UUID -DFMT_SHARED -DHAVE_CURL -DHAVE_INOTIFY -DHAVE_JS -DHAVE_NL_LANGINFO -DHAVE_SETLOCALE -DONLINE_SERVICES -DPACKAGE_DATADIR=\"/usr/local/share/gerbera\" -DPACKAGE_NAME=\"Gerbera\" -DSOPCAST -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -DUSING_NPUPNP -DVERSION=\"1.6.1\" -I/usr/local/include/npupnp -I/usr/ports/net/gerbera/work/gerbera-1.6.1/src -O2 -pipe -march=ivybridge -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -Wall -O2 -pipe -march=ivybridge -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -pthread -std=c++17 -MD -MT test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o -MF test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o.d -o test/util/CMakeFiles/testutil.dir/test_upnp_headers.cc.o -c /usr/ports/net/gerbera/work/gerbera-1.6.1/test/util/test_upnp_headers.cc
/usr/ports/net/gerbera/work/gerbera-1.6.1/test/util/test_upnp_headers.cc:13:16: error: use of undeclared identifier 'UpnpFileInfo_new'; did you mean 'UpnpFileInfo'?
        info = UpnpFileInfo_new();
               ^
/usr/local/include/npupnp/upnp.h:641:26: note: 'UpnpFileInfo' declared here
typedef struct File_Info UpnpFileInfo;
                         ^
/usr/ports/net/gerbera/work/gerbera-1.6.1/test/util/test_upnp_headers.cc:17:9: error: use of undeclared identifier 'UpnpFileInfo_delete'
        UpnpFileInfo_delete(info);
        ^
2 errors generated.
```

Gerbera v1.6.1 + commit e648763626e3c2512801bd127a0a3b96c8716faf
libnpupnp v4.0.11

FreeBSD 13-CURRENT, clang version 11.0.0 (git@github.com:llvm/llvm-project.git llvmorg-11.0.0-rc2-0-g414f32a9e86)

--- a/test/util/test_upnp_headers.cc
+++ b/test/util/test_upnp_headers.cc
@@ -10,12 +10,20 @@ public:
     HeadersHelperTest()
     {
         subject = new Headers();
+#if defined(USING_NPUPNP)
+        info = new UpnpFileInfo;
+#else
         info = UpnpFileInfo_new();
+#endif
     };
     virtual ~HeadersHelperTest() override
     {
+#if defined(USING_NPUPNP)
+        delete info;
+#else
         UpnpFileInfo_delete(info);
 
+#endif
         delete subject;
     };
 
@@ -23,6 +31,7 @@ public:
     Headers* subject;
 };
 
+#if !defined(USING_NPUPNP)
 TEST_F(HeadersHelperTest, TerminatesTheHeaderWithCarriageNewLine)
 {
     // arrange
@@ -261,3 +270,4 @@ TEST_F(HeadersHelperTest, HandlesExtraContentTwo)
     auto actual = Headers::readHeaders(info);
     EXPECT_EQ(*actual, expected);
 }
+#endif

